youtubearchive.py

```diff
@@ -0,0 +1,87 @@
+from dataclasses import dataclass
+import dataclasses
+import json
+import isodate
+import requests
+
+_api_base = "https://youtube.googleapis.com/youtube/v3"
+
+@dataclass
+class YoutubeVideoMetadata():
+    id: str
+    title: str
+    duration: int
+    description: str
+
+class YoutubeArchive():
+    _cache: list[YoutubeVideoMetadata]
+    _filename: str
+    _channel_id: str
+    _api_key: str
+
+    def __init__(self, filename: str, channel_id: str, api_key: str):
+        self._filename = filename
+        self._channel_id = channel_id
+        self._api_key = api_key
+        
+        self._reload_cache()
+
+    @property
+    def archive(self):
+        return self._cache
+
+    def _reload_cache(self):
+        if not os.path.isFile(self._filename):
+            return
+        
+        with open(self._filename, "r") as json_raw:
+            loaded = json.load(json_raw)
+
+            cache = []
+            for raw in loaded:
+                cache.append(YoutubeVideoMetadata(**raw))
+            self._cache = cache
+
+    # TODO: Setup calling this with TwitchIO?
```

Comment:
@Vgr255 By default the google developer API has 10k units to play with. This consumes 100 per page of new videos + 1 per new video. 

I think that means building this list from nothing will take less than 3000 units, then all subsequent runs should be ~101 units.

We could probably set this to run every hour or so very safely, and just forget about it?